code/game/objects/structures/flora.dm

```diff
@@ -73,14 +73,106 @@
 	..()
 	icon_state = "snowbush[rand(1, 6)]"
 
+
+
+
+//Potted Plants
 /obj/structure/flora/pottedplant
 	name = "potted plant"
 	icon = 'icons/obj/plants.dmi'
 	icon_state = "plant-26"
+	var/dead = 0
+	var/obj/item/stored_item
 	layer = ABOVE_MOB_LAYER
 
-//newbushes
 
+/obj/structure/flora/pottedplant/Destroy()
+	QDEL_NULL(stored_item)
+	return ..()
+
+/obj/structure/flora/pottedplant/proc/death()
+	if (!dead)
+		icon_state = "plant-dead"
+		name = "dead [name]"
+		desc = "It looks dead."
+		dead = 1
+//No complex interactions, just make them fragile
+/obj/structure/flora/pottedplant/ex_act(var/severity = 2.0)
+	death()
+	return ..()
+
+/obj/structure/flora/pottedplant/fire_act()
+	death()
+	return ..()
+
+/obj/structure/flora/pottedplant/attackby(obj/item/W, mob/user)
+	user.visible_message("[user] begins digging around inside of \the [src].", "You begin digging around in \the [src], trying to hide \the [W].")
+	if(do_after(user, 20, target = src))
+		if(!stored_item)
+			if(W.w_class <= 3)
+				user.drop_from_inventory(W,src)
+				stored_item = W
+				to_chat(user,"<span class='notice'>You hide \the [W] in [src].</span>")
+				return
+			else
+				to_chat(user,"<span class='notice'>\The [W] can't be hidden in [src], it's too big.</span>")
+				return
+		else
+			to_chat(user,"<span class='notice'>There is something hidden in [src].</span>")
+			return
+	return ..()
+
+/obj/structure/flora/pottedplant/attack_hand(mob/user)
+	user.visible_message("[user] begins digging around inside of \the [src].", "You begin digging around in \the [src], searching it.")
+	if(do_after(user, 40, target = src))
+		if(!stored_item)
+			to_chat(user,"<span class='notice'>There is nothing hidden in [src].</span>")
+		else
+			if(istype(stored_item, /obj/item/device/paicard))
+				stored_item.forceMove(src.loc)
+				to_chat(user,"<span class='notice'>You reveal \the [stored_item] from [src].</span>")
+			else
+				user.put_in_hands(stored_item)
+				to_chat(user,"<span class='notice'>You take \the [stored_item] from [src].</span>")
+			stored_item = null
+
+/obj/structure/flora/pottedplant/bullet_act(var/obj/item/projectile/Proj)
+	if (prob(Proj.damage*2))
+		death()
+		return 1
+	return ..()
+
+
+//Added random icon selection for potted plants.
+//It was silly they always used the same sprite when we have 26 sprites of them in the icon file
+/obj/structure/flora/pottedplant/random
+	//Since Eris has cyberplants, we will have these random plants have a chance to become a cyberplant instead
+	var/cyber_chance = 20
+
+/obj/structure/flora/pottedplant/random/Initialize()
+	..()
+	if (prob(cyber_chance))
+		new /obj/structure/cyberplant(loc)
+		qdel(src)
+		return
```

Comment:
Why `qdel(src); return` instead `return INITIALIZE_HINT_QDEL`?

Reply:
i'd never heard of this feature until now. change made, and it seems to work well. thanks for the info!